setup/steps/enable_auto_explain.go

```diff
@@ -0,0 +1,59 @@
+package steps
+
+import (
+	"errors"
+	"strings"
+
+	survey "github.com/AlecAivazis/survey/v2"
+	s "github.com/pganalyze/collector/setup/state"
+	"github.com/pganalyze/collector/setup/util"
+)
+
+var EnableAutoExplain = &s.Step{
+	Kind:        s.AutomatedExplainStep,
+	Description: "Enable auto_explain",
+	Check: func(state *s.SetupState) (bool, error) {
+		logExplain, err := util.UsingLogExplain(state.CurrentSection)
+		if err != nil || logExplain {
+			return logExplain, err
+		}
+		spl, err := util.GetPendingSharedPreloadLibraries(state.QueryRunner)
+		if err != nil {
+			return false, err
+		}
+		return strings.Contains(spl, "auto_explain"), nil
+	},
+	Run: func(state *s.SetupState) error {
+		var doAdd bool
+		if state.Inputs.Scripted {
+			if !state.Inputs.EnableAutoExplain.Valid || !state.Inputs.EnableAutoExplain.Bool {
+				return errors.New("enable_auto_explain flag not set but auto_explain configuration selected")
+			}
+			doAdd = state.Inputs.EnableAutoExplain.Bool
+		} else {
+			err := survey.AskOne(&survey.Confirm{
+				Message: "Add auto_explain to shared_preload_libraries (will be saved to Postgres)?",
```

Comment:
`Add auto_explain to shared_preload_libraries (will be saved to Postgres - requires a restart)?`